buildSrc/src/main/groovy/org/elasticsearch/gradle/VersionProperties.java

```diff
@@ -0,0 +1,50 @@
+package org.elasticsearch.gradle;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Accessor for shared dependency versions used by elasticsearch, namely the elasticsearch and lucene versions.
+ */
+public class VersionProperties {
```

Comment:
It appears this is being moved to a java file, but it is still under `src/main/groovy`?

Reply:
Yes. That is because the Groovy compiler knows how to deal with this and delegate these files to the Java compiler. The reason these can't move to `src/main/java` (yet) us that they make use of Groovy classes that are not on the cp  when pure java is compiled. Moving those Groovy classes to Java creates a ripple effect, so my thinking is that we can have java files in the groovy source set until we have critical mass, eventually removing the groovy source set. It's easier and less risky than converting to Java at once.

Reply:
This sounds like a great plan!